src/CorrelationId/Abstractions/ICorrelationIdProvider.cs

```diff
@@ -12,6 +13,6 @@ public interface ICorrelationIdProvider
         /// </summary>
         /// <param name="context">The <see cref="HttpContext"/> of the current request.</param>
         /// <returns>A string representing the correlation ID.</returns>
-        string GenerateCorrelationId(HttpContext context);
+        Task<string> GenerateCorrelationId(HttpContext context);
```

Comment:
Since the method is now `async`, it should probably be named `GenerateCorrelationIdAsync`

Reply:
What are your thoughts on accepting a `CancellationToken` as well?